design/proposals/ambient-namespaces.md

```diff
@@ -0,0 +1,52 @@
+# Ambient Namespaces
+
+Status: Pending
+
+Version: Alpha
+
+## Abstract
+
+We want to introduce functionality to enable the use of environment metadata (ex: cluster namespace, Server URI) in prototype and component manifest files.
+
+## Motivation
+
+* Environments have configured metadata, such as the cluster’s namespace or server URI. Users [have requested](https://github.com/ksonnet/ksonnet/issues/222) that these fields be referenceable in both prototypes and component files.
+* Component and prototype manifests should not reference a specific environment's details. See Goals and Non Goals for more clarity on what "specific" implies.
```

Comment:
I think the first of these is pretty good motivation, and you substantiate it well. The second is really more of a constraint and you might just defer that to the goals/non-goals.